src/integrationtests/java/com/aws/greengrass/integrationtests/status/ComponentStatusChangeFleetStatusServiceTest.java

```diff
@@ -0,0 +1,202 @@
+        fleetStatusDetails = fleetStatusDetailsList.get().get(1);
+        assertEquals(Trigger.COMPONENT_STATUS_CHANGE, fleetStatusDetails.getTrigger());
+        assertTrue(fleetStatusDetails.getComponentDetails().stream()
+                .anyMatch(c -> "ServiceB".equals(c.getComponentName()) && State.ERRORED.equals(c.getState())));
+
+        fleetStatusDetails = fleetStatusDetailsList.get().get(2);
+        assertEquals(Trigger.COMPONENT_STATUS_CHANGE, fleetStatusDetails.getTrigger());
+        assertTrue(fleetStatusDetails.getComponentDetails().stream()
+                .anyMatch(c -> "ServiceA".equals(c.getComponentName()) && State.ERRORED.equals(c.getState())));
+
+        // the fourth message should be component recovery
+        fleetStatusDetails = fleetStatusDetailsList.get().get(3);
+        assertEquals(Trigger.COMPONENT_STATUS_CHANGE, fleetStatusDetails.getTrigger());
+        assertTrue(fleetStatusDetails.getComponentDetails().stream()
+                .anyMatch(c -> "ServiceA".equals(c.getComponentName()) && State.RUNNING.equals(c.getState())));
+        assertTrue(fleetStatusDetails.getComponentDetails().stream()
+                .anyMatch(c -> "ServiceB".equals(c.getComponentName()) && State.RUNNING.equals(c.getState())));
+    }
+    @Test
+    void GIVEN_three_components_errored_and_recovered_THEN_fss_should_send_only_one_recovery_message_and_one_errored_message_should_contain_recovery_message() throws Exception {
+        statusChange  = new CountDownLatch(4);
+        ConfigPlatformResolver.initKernelWithMultiPlatformConfig(kernel,
+                ComponentStatusChangeFleetStatusServiceTest.class.getResource("fss_service_10s_recovery.yaml"));
+        kernel.launch();
+
+        FleetStatusService fss = (FleetStatusService) kernel.locateIgnoreError(FLEET_STATUS_SERVICE_TOPICS);
+        fss.setWaitBetweenPublishDisabled(true);
+
+        assertTrue(statusChange.await(30, TimeUnit.SECONDS));
+        // we expect a total of 5 messages, 1 Nucleus launch, 4 component status change includes:
+        // 1 Errored from A with B reovered, 1 Errored B, 1 Errored C, 1 recovery message for the rest of non recovery ones
+        assertEquals(5, fleetStatusDetailsList.get().size());
+
+        // the first message should be nucleus launch
+        FleetStatusDetails fleetStatusDetails = fleetStatusDetailsList.get().get(0);
+        assertEquals(OverallStatus.HEALTHY, fleetStatusDetails.getOverallStatus());
+        assertEquals(Trigger.NUCLEUS_LAUNCH, fleetStatusDetails.getTrigger());
+
+        // the second should be errored component C
+        fleetStatusDetails = fleetStatusDetailsList.get().get(1);
+        assertEquals(Trigger.COMPONENT_STATUS_CHANGE, fleetStatusDetails.getTrigger());
+        assertTrue(fleetStatusDetails.getComponentDetails().stream()
+                .anyMatch(c -> "ServiceC".equals(c.getComponentName()) && State.ERRORED.equals(c.getState())));
```

Comment:
nit: can we refactor this into a function? like `assertComponentStatus`

Reply:
Updated